### PR TITLE
Bump Go version to 1.22.10

### PR DESCRIPTION
**Pull request by deanurag**
Updated the Go Version to 1.22.10 from 1.22.9.

Part of https://github.com/etcd-io/etcd/issues/19000

### CODE REVIEW

File: integration/sleep/go.mod [hunk 1]
```diff
@@ -2,7 +2,7 @@ module go.etcd.io/gofail/integration/sleep
 
 go 1.22
 
-toolchain go1.22.9
+toolchain go1.22.10S
```
Comment (ivanvc) [inline integration/sleep/go.mod:5]: There's a trailing "S" in this line.

Comment (deanurag) [reply integration/sleep/go.mod]: Extemely sorry, for such mistake. I will do the required changes ASAP.

**Comment by deanurag** [comment]
@ivanvc Made the changes, please take a look.

**Comment by ivanvc** [comment]
@deanurag, can you please squash your commits into a single one? You can do this with:

```
git reset --soft HEAD^
git commit --amend --signoff -m "Update Go version from 1.22.9 to 1.22.10"
git push --force
```

Thanks :)

**Comment by ivanvc** [comment]
@ahrtr, I think this needs manual workflow approval.

**Pull request merged**